_posts/2020-04-23-putting-the-serverless-in-graphQL-with-AppSync.md

```diff
@@ -0,0 +1,206 @@
+
+```yml
+CoffeeShopDynamoDB:
+  Type: AWS::DynamoDB::Table
+  Properties:
+    TableName: "CoffeeShopTable"
+    AttributeDefinitions:
+      - AttributeName: "id"
+        AttributeType: "S"
+    KeySchema:
+      - AttributeName: "id"
+        KeyType: "HASH"
+    ProvisionedThroughput:
+      ReadCapacityUnits: 3
+      WriteCapacityUnits: 1
+    Projection:
+      ProjectionType: "ALL"
+
+CoffeeShopDynamoDBDataSource:
+  Type: AWS::AppSync::DataSource
+  Properties:
+    ApiId: !GetAtt CoffeeShopAppSyncApi.ApiId
+    Name: CoffeeShopDynamoDBDataSource
+    Description: Contain order data from coffee shop
+    Type: “AMAZON_DYNAMODB”
+    ServiceRoleArn: arn:aws:iam::xxxxxxxxxxxx:role/xxxxxxxxx
+    DynamoDBConfig:
+      TableName: CoffeeShopTable
+      AwsRegion: ap-southeast-2
+```
+
+From the template, we are provisioning new DynamoDB table for the API. It has the attribute id as the hash key and resourceId as the range key for the table.
+
+We are connecting the AppSync data source with the newly created DynamoDB table by specifying the table name. You can connect to an existing data source if you already have one.
+
+[Import a DynamoDB Table](https://docs.aws.amazon.com/appsync/latest/devguide/import-dynamodb.html#import-a-ddb-table)
+
+### Step 4: Create the Resolver
+
+Adding a resolver for a mutation, query or subscription allows us to map data from our client to our data source, in this case DynamoDB, and vice versa.
+
+```yml
+CreateOrderResolver:
+  Type: AWS:: AppSync::Resolver
+  Properties:
+    ApiId: !GetAtt CoffeeShopAppSyncApi.ApiId
+    TypeName: Mutation
+    FieldName: createOrder
+    DataSourceName: !GetAtt CoffeeShopDynamoDBDataSource.Name
+    RequestMappingTemplate: |
+      {
+        "version" : "2017-02-28",
+        "operation" : "PutItem",
+        "key" : {
+          "id": { "S" : "${ctx.args.orderInput.id}" },
+          "resourceId": { "S": "COFFEE" }
+        },
+      }
+    ResponseMappingTemplate: |
+        $util.toJson($context.result)
+
+GetOrderResolver:
+  Type: AWS:: AppSync::Resolver
+  Properties:
+    ApiId: !GetAtt CoffeeShopAppSyncApi.ApiId
+    TypeName: Query
+    FieldName: getOrder
+    DataSourceName: !GetAtt CoffeeShopDynamoDBDataSource.Name
+    RequestMappingTemplate: |
+      {
+          "version": "2017-02-28",
+          "operation": "GetItem",
+          "key": {
+            "id": { "S": "${ctx.args.id}" },
+            "resourceId": { "S": "COFFEE" }
+          }
+        }
+    ResponseMappingTemplate: |
+        $util.toJson($context.result)
+```
+
+# Conclusion
+
+We had a brief introduction on how you can provision AWS AppSync with DynamoDb as the data source by using CloudFormation. AWS has pretty good documentation, it is worth reading if you want to get more details.
+
+[AWS APPSYNC](https://aws.amazon.com/appsync/)
+
+Overall I think that AWS AppSync is a good starting point for developers who want to start building an API for their application.
```

Comment:
why?

Reply:
Why, and this should go in the intro.

In the conclusion, just recap what you taught.